src/ln/peer_handler.rs

```diff
@@ -576,6 +586,7 @@ impl<Descriptor: SocketDescriptor> PeerManager<Descriptor> {
 	/// calls to ChannelManager::process_pending_htlc_forward.
 	pub fn process_events(&self) {
 		let mut upstream_events = Vec::new();
+		let mut disconnect_peers = Vec::new();
```

Comment:
I dont see why you need this - you're doing two map lookups, passing an event to that entry, then doing another two map lookups to remove it, might as well replace the get_peer_for_forwarding with a written out version that removes the entries all at once.